app/containers/SearchContainer/index.js

```diff
@@ -0,0 +1,100 @@
+/**
+ *
+ * SearchContainer
+ *
+ */
+
+import React, { memo, useEffect } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { injectIntl, FormattedMessage as T } from 'react-intl';
+import { createStructuredSelector } from 'reselect';
+import { compose } from 'redux';
+import { injectSaga } from 'redux-injectors';
+import { selectSearchContainer, selectTrackData, selectTrackError, selectSearchTerm } from './selectors';
+import searchContainerSaga from './saga';
+import { Input } from 'antd';
+import get from 'lodash/get';
+import { isEmpty } from 'lodash';
+import debounce from 'lodash/debounce';
+import { searchContainerCreators } from './reducer';
+
+const { Search } = Input;
+
+export function SearchContainer({
+  dispatchTrackInfo,
+  dispatchClearTrackInfo,
+  intl,
+  searchTerm,
+  trackError,
+  trackData
+}) {
+  useEffect(() => {
+    get(trackData) || trackError;
+  }, [trackData]);
+
+  useEffect(() => {
+    if (searchTerm && !trackData) {
+      dispatchTrackInfo(searchTerm);
+    }
+  });
+
+  const handleOnChange = (sTerm) => {
+    if (!isEmpty(sTerm)) {
+      dispatchTrackInfo(sTerm);
```

Comment:
for creators for API calls we use the following convention

`<request|success|failure><http-method><entity>`

and then the dispatchers would contain the above prefixed with `dispatch`. 

It makes it a bit more readable and we know exactly what the underlying API is